modules/nextflow/src/main/groovy/nextflow/ast/NextflowDSLImpl.groovy

```diff
@@ -640,8 +641,9 @@ class NextflowDSLImpl implements ASTTransformation {
                             readSource(stm,source,unit)
                             break
 
-                        case 'script':
                         case 'shell':
+                            log.warn1 "Process $name > the `shell` block is deprecated, use `script` instead"
```

Comment:
Minor improvement, instead of reporting this warning here, it could be done at TaskProcessor creation level, this guarantee that's called once and the process name is provided 

https://github.com/nextflow-io/nextflow/blob/c6dfef6188127e1b036923f1304795de57b38071/modules/nextflow/src/main/groovy/nextflow/processor/TaskProcessor.groovy#L311-L311

Reply:
done